File: CItem_proj_2/citem.py
from typing import List

class CItem:
    ACCEPTABLE_DIRECTIONS: List[str] = ["left", "right", "down", "none"]

    def __init__(self, name: str, acquired: bool = False, cApplied: str = "none", inUse: bool = False):
        self._name = name
        self._acquired = acquired
        if cApplied not in self.ACCEPTABLE_DIRECTIONS or self._acquired == False:
            self._cApplied = "none"
        else:
            self._cApplied = cApplied
        if self._cApplied == "none":
            self._inUse = False
        else:
            self._inUse = inUse
    
    @property
    def name(self) -> str:
        return self._name

    @property
    def acquired(self) -> bool:
        return self._acquired

    def beenAcquired(self) -> None:
        self._acquired = True
    
    @property
    def cApplied(self) -> str:
        return self._cApplied

    def applyC(self, newDirect: str) -> None:
        if newDirect not in self.ACCEPTABLE_DIRECTIONS or self._acquired == False:
            self._cApplied = "none"
        else:
            self._cApplied = newDirect
        if self._cApplied == "none":
            self._inUse = False

    @property
    def inUse(self) -> bool:
        return self._inUse
    
    def using(self) -> None:
        if self._cApplied != "none":
            self._inUse = True

    def stopUsing(self) -> None:
        self._inUse = False

File: CItem_proj_2_cpp/src/CItem.cpp
#include "CItem.h"

namespace item {
    const std::string CItem::ACCEPTABLE_DIRECTIONS[4] = {"none", "left", "right", "down"};

    CItem::CItem(std::string _name, bool _acquired, std::string _cApplied, bool _inUse)
        : name(_name), acquired(_acquired), cApplied(_cApplied), inUse(_inUse)
    {
        if(acquired == false) {
            cApplied = "none";
            inUse = false;
        }
        bool accept = false;
        for(int i = 0; i < 4; i++) {
            if(cApplied == ACCEPTABLE_DIRECTIONS[i]) accept = true;
        }
        if(!accept) cApplied = "none";

        if(cApplied == "none") inUse = false;        
    }

    std::string CItem::getName() const{
        return name;
    }

    void CItem::beenAcquired() {
        acquired = true;
    }

    bool CItem::getAcquired() const{
        return acquired;
    }

    std::string CItem::getCDirection() const {
        return cApplied;
    }

    void CItem::setCDirection(std::string newDirect) {
        bool accept = false;
        for(int i = 0; i < 4; i++) {
            if(newDirect == ACCEPTABLE_DIRECTIONS[i]) accept = true;
        }
        if(!accept) cApplied = "none";
        else cApplied = newDirect;
        if(cApplied == "none") inUse = false;
    }

    bool CItem::getUsing() const{
        return inUse;
    }

    void CItem::useItem() {
        bool accept = false;
        for(int i = 0; i < 4; i++) {
            if(cApplied == ACCEPTABLE_DIRECTIONS[i]) accept = true;
        }
        if(accept && acquired && cApplied != "none") inUse = true;
    }

    void CItem::stopUsing() {
        inUse = false;
    }
}